assets/src/edit-story/components/canvas/pagemenu/pageMenu.js

```diff
@@ -0,0 +1,186 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import styled from 'styled-components';
+import { useCallback } from 'react';
+import { __, sprintf } from '@web-stories-wp/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { Icons } from '../../../../design-system';
+import { useStory, useHistory, useConfig, useCanvas } from '../../../app';
+import { createPage, duplicatePage } from '../../../elements';
+import PageMenuButton from './pageMenuButton';
+import AnimationToggle from './animationToggle';
+
+const DIVIDER_HEIGHT = 16;
+
+const Wrapper = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+
+  --pagemenu-icon-space: ${({ isWidePage }) => (isWidePage ? 16 : 10)}px;
+  --pagemenu-count-space: ${({ isWidePage }) => (isWidePage ? 24 : 12)}px;
+`;
+
+const PageCount = styled.div`
+  color: ${({ theme }) => theme.colors.fg.primary};
+  font-family: ${({ theme }) => theme.DEPRECATED_THEME.fonts.body1.family};
+  font-size: ${({ theme }) => theme.DEPRECATED_THEME.fonts.body1.size};
+  line-height: 24px;
+`;
+
+const Divider = styled.span`
+  background-color: ${({ theme }) => theme.colors.bg.tertiary};
+  opacity: 0.3;
+  height: ${DIVIDER_HEIGHT}px;
+  width: 1px;
+`;
+
+const CountSpace = styled.div`
+  width: var(--pagemenu-count-space);
+`;
+
+const IconSpace = styled.div`
+  width: var(--pagemenu-icon-space);
+`;
+
+function PageMenu() {
+  const {
+    state: { canUndo, canRedo },
+    actions: { undo, redo },
+  } = useHistory();
+  const {
+    currentPageNumber,
+    currentPage,
+    deleteCurrentPage,
+    addPage,
+    hasAnimations,
+  } = useStory(
+    ({
+      state: { currentPageNumber, currentPage },
+      actions: { deleteCurrentPage, addPage },
+    }) => {
+      return {
+        currentPageNumber,
+        currentPage,
+        deleteCurrentPage,
+        addPage,
+        hasAnimations: currentPage?.animations?.length > 0,
+      };
+    }
+  );
+  const { pageSize } = useCanvas((state) => ({
+    pageSize: state.state.pageSize,
+  }));
+  const { isRTL } = useConfig();
+
+  const handleDeletePage = useCallback(() => deleteCurrentPage(), [
+    deleteCurrentPage,
+  ]);
+
+  const handleAddPage = useCallback(() => addPage({ page: createPage() }), [
+    addPage,
+  ]);
+
+  const handleDuplicatePage = useCallback(
+    () => addPage({ page: duplicatePage(currentPage) }),
+    [addPage, currentPage]
+  );
+
+  const isWidePage = pageSize.width > 280;
```

Comment:
280 to a constant perhaps?

Reply:
It's temporary only until zoom is merged in - then this will change to some actual calculated value.